yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1233,6 +1298,51 @@ def pixelize_sph_kernel_arbitrary_grid(np.float64_t[:, :, :] buff,
 
                         buff[xi, yi, zi] += coeff * kernel_func(q)
 
+@cython.initializedcheck(False)
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def pixelize_sph_kernel_gather_arbitrary_grid(np.float64_t[:, :, :] buff,
+        np.int64_t[:, :, :, :] pids, np.float64_t[:, :, :, :] dists,
+        np.float64_t[:] hsml, np.float64_t[:] pmass, np.float64_t[:] pdens,
+        np.float64_t[:] quantity_to_smooth, PyKDTree tree=None, np.int64_t
+        offset=0, kernel_name="cubic"):
+
+    cdef np.intp_t xsize, ysize, zsize
+    cdef np.float64_t w_j, coeff
+    cdef np.int64_t xi, yi, zi, pi
+    cdef np.float64_t q, h_j2, ih_j2
+    cdef int count, i, j, k, particle
+    cdef np.int64_t[:] tree_id
+
+    if tree is not None:
+        tree_id = tree.idx.astype("int64")
+
+    xsize, ysize, zsize = buff.shape[0], buff.shape[1], buff.shape[2]
+
+    kernel_func = get_kernel_func(kernel_name)
+
+    with nogil:
+        for xi in range(xsize):
+            for yi in range(ysize):
+                for zi in range(zsize):
+                    h_j2 = dists[xi, yi, zi, 0]
```

Comment:
Is this the smoothing length of j squared?

Reply:
Yeah, despite my ambiguous naming the bounded priority queue actually stores the distances squared.

Reply:
Okay, it might be worth making that a little clearer here. This is the _distance_ squared? _h_ is usually reserved for smoothing length -- I would use the notation r_ij_2 (normally we talk about interacting particle "i", yourself, with particle "j", another particle).

Reply:
It is a smoothing length squared. We are saying that the smoothing length squared is equal to distance squared to the furthers neighbor in our list.

Reply:
Great, I would like to have a few more comments here (or more descriptive variable names) as I mentioned in my other comment -- but other than that this looks good.

Reply:
Now you've highlighted this region - I can see how that particular line can be confusing, it's a bit arbitrary. I'll add some more comments to this section.